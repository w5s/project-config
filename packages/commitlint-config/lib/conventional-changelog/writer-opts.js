"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var node_fs_1 = require("node:fs");
var node_path_1 = require("node:path");
var transform_js_1 = require("./transform.js");
var basePath = (0, node_path_1.resolve)(__dirname, './templates');
var template = (0, node_fs_1.readFileSync)("".concat(basePath, "/template.hbs"), 'utf8');
var header = (0, node_fs_1.readFileSync)("".concat(basePath, "/header.hbs"), 'utf8');
var commit = (0, node_fs_1.readFileSync)("".concat(basePath, "/commit.hbs"), 'utf8');
var footer = (0, node_fs_1.readFileSync)("".concat(basePath, "/footer.hbs"), 'utf8');
var author = (0, node_fs_1.readFileSync)("".concat(basePath, "/author.hbs"), 'utf8');
var writerOpts = {
    transform: (0, transform_js_1.createTransform)({}),
    groupBy: 'type',
    commitGroupsSort: 'title',
    commitsSort: ['scope', 'subject'],
    noteGroupsSort: 'title',
    mainTemplate: template,
    headerPartial: header,
    commitPartial: commit.replace(/{{gitUserInfo}}/g, author),
    footerPartial: footer,
};
exports.default = writerOpts;
//# sourceMappingURL=writer-opts.js.map