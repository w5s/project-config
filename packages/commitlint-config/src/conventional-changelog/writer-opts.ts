import type { Options } from 'conventional-changelog-writer';
import { readFileSync } from 'node:fs';
import { resolve } from 'node:path';
import { createTransform } from './transform.js';

const basePath = resolve(__dirname, './templates');

const template = readFileSync(`${basePath}/template.hbs`, 'utf8');
const header = readFileSync(`${basePath}/header.hbs`, 'utf8');
const commit = readFileSync(`${basePath}/commit.hbs`, 'utf8');
const footer = readFileSync(`${basePath}/footer.hbs`, 'utf8');
const author = readFileSync(`${basePath}/author.hbs`, 'utf8');

const writerOpts: Options = {
  transform: createTransform({}),
  groupBy: 'type',
  commitGroupsSort: 'title',
  commitsSort: ['scope', 'subject'],
  noteGroupsSort: 'title',
  mainTemplate: template,
  headerPartial: header,
  commitPartial: commit.replace(/{{gitUserInfo}}/g, author),
  footerPartial: footer,
};

export default writerOpts;
