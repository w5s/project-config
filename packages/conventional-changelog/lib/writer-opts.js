"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.writerOpts = exports.defaultDisplayTypes = void 0;
const node_fs_1 = require("node:fs");
const node_path_1 = require("node:path");
const transform_js_1 = require("./transform.js");
const data_js_1 = require("./data.js");
const basePath = (0, node_path_1.resolve)((0, node_path_1.dirname)(__dirname), './template');
const template = (0, node_fs_1.readFileSync)(`${basePath}/template.hbs`, 'utf8');
const header = (0, node_fs_1.readFileSync)(`${basePath}/header.hbs`, 'utf8');
const commit = (0, node_fs_1.readFileSync)(`${basePath}/commit.hbs`, 'utf8');
const footer = (0, node_fs_1.readFileSync)(`${basePath}/footer.hbs`, 'utf8');
const author = (0, node_fs_1.readFileSync)(`${basePath}/author.hbs`, 'utf8');
exports.defaultDisplayTypes = data_js_1.CommitConventionalType.findWhere((_) => _.changelog);
exports.writerOpts = {
    transform: (0, transform_js_1.createTransform)({
        displayTypes: exports.defaultDisplayTypes,
    }),
    groupBy: 'type',
    commitGroupsSort: 'title',
    commitsSort: ['scope', 'subject'],
    noteGroupsSort: 'title',
    mainTemplate: template,
    headerPartial: header,
    // eslint-disable-next-line unicorn/prefer-string-replace-all
    commitPartial: commit.replace(/{{gitUserInfo}}/g, author),
    footerPartial: footer,
};
