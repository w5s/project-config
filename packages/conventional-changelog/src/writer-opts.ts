import type { Options } from 'conventional-changelog-writer';
import { readFileSync } from 'node:fs';
import { resolve, dirname } from 'node:path';
import { createTransform } from './transform.js';

export interface WriterOptions extends Options {}

const basePath = resolve(dirname(__dirname), './template');

const template = readFileSync(`${basePath}/template.hbs`, 'utf8');
const header = readFileSync(`${basePath}/header.hbs`, 'utf8');
const commit = readFileSync(`${basePath}/commit.hbs`, 'utf8');
const footer = readFileSync(`${basePath}/footer.hbs`, 'utf8');
const author = readFileSync(`${basePath}/author.hbs`, 'utf8');

export const writerOpts: WriterOptions = {
  transform: createTransform({}),
  groupBy: 'type',
  commitGroupsSort: 'title',
  commitsSort: ['scope', 'subject'],
  noteGroupsSort: 'title',
  mainTemplate: template,
  headerPartial: header,
  commitPartial: commit.replace(/{{gitUserInfo}}/g, author),
  footerPartial: footer,
};
