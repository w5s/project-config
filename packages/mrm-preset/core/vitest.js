const fs = require('node:fs');
const { file, packageJson } = require('mrm-core');
const pkg = require('./pkg.js');
const npm = require('./npm.js');
const project = require('./project.js');

/**
 * @param {{
 *   state: 'present'|'absent'
 * }} options
 */
function vitest({ state }) {
  const hasVitest = state === 'present';
  const hasWorkspace = pkg.hasWorkspaces(packageJson());
  const vitestScript = 'vitest run';
  const update = (/** @type {string|undefined} */ _) =>
    hasVitest ? vitestScript : _ === vitestScript ? pkg.emptyScript : _;

  // Dependencies
  npm.dependency({
    dev: true,
    name: ['vite', 'vitest'],
    state: hasVitest ? 'present' : 'absent',
  });

  /** @type {string|undefined} */
  let viteVersion;
  /** @type {string|undefined} */
  let vitestVersion;
  pkg.withPackageJson((packageFile) => {
    viteVersion = packageFile.get('devDependencies.vite');
    vitestVersion = packageFile.get('devDependencies.vitest');
    pkg.script(packageFile, {
      name: `${project.test}:root`,
      update,
      state: hasWorkspace ? 'absent' : 'present',
    });
  });

  pkg.forEachWorkspace(({ projectDir, packageFile }) => {
    const hasSrc = fs.existsSync(`${projectDir}/src`);
    const packageState = hasVitest && hasSrc ? 'present' : 'absent';
    pkg.value(packageFile, {
      path: 'devDependencies.vite',
      state: packageState,
      update: (_) => _ ?? viteVersion,
    });
    pkg.value(packageFile, {
      path: 'devDependencies.vitest',
      state: packageState,
      update: (_) => _ ?? vitestVersion,
    });
    pkg.script(packageFile, {
      name: `${project.test}:src`,
      update,
      state: packageState,
    });
    const viteConfig = file(`${projectDir}/vite.config.mts`);
    if (packageState === 'absent') {
      viteConfig.delete();
    } else if (!viteConfig.exists()) {
      viteConfig.save(
        `/// <reference types="vitest" />
import { defineConfig } from 'vite';

export default defineConfig({
  test: {},
});
`
      );
    }
  });
}

module.exports = {
  vitest,
};
