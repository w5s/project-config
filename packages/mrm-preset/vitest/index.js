const { vitest } = require('../core/vitest.js');

/**
 *
 */
function task() {
  vitest({
    state: 'present',
  });
}

task.description = 'Setup Vitest';
task.parameters = {};

module.exports = task;
