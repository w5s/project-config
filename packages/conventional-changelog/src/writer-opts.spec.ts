import { describe, expect, it } from 'vitest';
import { defaultDisplayTypes } from './writer-opts';

describe('defaultDisplayTypes', () => {
  it('should filter types with changelog=true', () => {
    expect(defaultDisplayTypes).toMatchSnapshot();
  });
});
