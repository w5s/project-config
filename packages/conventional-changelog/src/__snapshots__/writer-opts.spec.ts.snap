// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaultDisplayTypes > should be {format} 1`] = `
[
  "feat",
  "fix",
  "perf",
  "revert",
]
`;
